source/docs/software/sensors/limit-switch.rst

```diff
@@ -0,0 +1,51 @@
+Programming Limit Switches
+==========================
+
+Limit switches are often used to control mechanisms on robots. While limit switches are simple to use, they only can sense a single position of a moving part. This makes them ideal for ensuring that movement doesn't exceed some limit but not so good at controlling the speed of the movement as it approaches the limit. For example, a rotational shoulder joint on a robot arm would best be controlled using a potentiometer or an absolute encoder, the limit switch could make sure that if the potentiometer ever failed, the limit switch would stop the robot from going to far and causing damage.
+
+Values Provided
+---------------
+
+.. image:: images/using-limit-switches/image1.png
+
+Limit switches can have "normally opened" or "normally closed" outputs. Normally open means that the switch will report low (0 value) until the switch is closed and then it will output high (1 value).  Normally closed works in the opposite way (high until closed then low).  The usual way of wiring the switch is between a digital input signal connection and ground. The digital input has pull-up resistors that will make the input be high (1 value) when the switch is open, but when the switch closes the value goes to 0 since the input is now connected to ground. The switch shown here has both normally open and normally closed outputs.  To learn more about limit switch hardware see this :ref:`article <docs/hardware/sensors/proximity-switches:Mechanical Proximity Switches ("limit switches")>`.
```

Comment:
The second sentence of doesn't match the rest of the paragraph here.  As stated later in the paragraph, a NO switch hooked up between the Rio signal and ground will report high (1 value) until the switch is closed and then it will output low (0 value).  The second and third sentences probably shouldn't use high/low at all, but rather talk about connected or disconnected.

Reply:
So I think first off I will move all of this over to the hardware section.  I think we should probably keep this to programming and link over there.  It already has some NO / NC stuff that might make this moot.  So when you get a second can you take a look at the new changes and tell me if you agree.

Reply:
The changes look good, thanks.